File: android/app/src/main/java/com/HammersTech/RoutineChart/core/domain/repositories/RoutineStepRepository.kt
package com.HammersTech.RoutineChart.core.domain.repositories

import com.HammersTech.RoutineChart.core.domain.models.RoutineStep
import kotlinx.coroutines.flow.Flow

/**
 * Repository interface for RoutineStep operations
 */
interface RoutineStepRepository {
    suspend fun create(step: RoutineStep)

    suspend fun update(step: RoutineStep)

    suspend fun getById(id: String): RoutineStep?

    suspend fun getByRoutineId(routineId: String): List<RoutineStep>

    fun observeByRoutineId(routineId: String): Flow<List<RoutineStep>>
    // Note: getByFamilyId removed - RoutineStep no longer has familyId
    // Steps are accessed via their parent Routine (getByRoutineId)
}

File: ios/RoutineChart/RoutineChart/Core/Domain/Repositories/RoutineStepRepository.swift
//
//  RoutineStepRepository.swift
//  RoutineChart
//
//  Repository protocol for RoutineStep data access
//

import Foundation

protocol RoutineStepRepository {
    func create(_ step: RoutineStep) async throws
    func get(id: String) async throws -> RoutineStep?
    func update(_ step: RoutineStep) async throws
    func getAll(routineId: String) async throws -> [RoutineStep]
    func softDelete(id: String) async throws
}

